### PR TITLE
Add new gpt-4-turbo 128k model (gpt-4-1106-preview) to cost and token…

### PR DESCRIPTION
**Pull request by hesiod-au**
… count logic (makes model usable)

### CODE REVIEW

File: mentat/llm_api.py [hunk 1]
```diff
@@ -160,7 +160,9 @@ def is_model_available(model: str) -> bool:
 
 
 def model_context_size(model: str) -> Optional[int]:
-    if "gpt-4" in model:
+    if model == "gpt-4-1106-preview":
+        return 131072
```
Comment (jakethekoenig) [inline mentat/llm_api.py:164]: Where do you see this context size? I see [128000](https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo) but maybe they announced a more precise number elsewhere?

**Review by jakethekoenig — COMMENTED** [review]
Thanks for doing this! One quick question.

File: mentat/llm_api.py (hunk 1 above)
Comment (hesiod-au) [reply mentat/llm_api.py:164]: I was applying their previous definition of "K" but as they have documented 128K as 128,000 I will change this.

**Review by jakethekoenig — APPROVED** [review]

**Comment by hesiod-au** [comment]
Thanks for the quick review!

**Comment by jochenstu** [comment]
Maybe related, using this on large requests I am getting 

```Error accessing OpenAI API: Request too large for gpt-4-0314 in organization * on tokens per min (TPM): Limit 10000, Requested 19102. The input or output tokens must be reduced in order to run successfully. Visit https://platform.openai.com/account/rate-limits to learn more.```

```
{
  "model": "gpt-4-1106-preview",
  "maxium-context": 128000,
  "auto_context": "true",
}
```

Is it actually running the new model for you?

**Comment by jakethekoenig** [comment]
> Maybe related, using this on large requests I am getting 
> 
> `
> Error accessing OpenAI API: Request too large for gpt-4-0314 in organization * on tokens per min (TPM): Limit 10000, Requested 19102. The input or output tokens must be reduced in order to run successfully. Visit https://platform.openai.com/account/rate-limits to learn more.
> `
> 
> ```
> {
>   "model": "gpt-4-1106-preview",
>   "maxium-context": 128000,
>   "auto_context": "true",
> }
> ```
> 
> Is it actually running the new model for you?
> 

Your config example has `maximum-context` misspelled and is invalid json (trailing comma). Do you not get a warning for it on mentat start up? Maybe it's in the wrong place?

Also note you can now change the model (or any other setting) with `/config model gpt-4-1106-preview`. I need to update the docs.

**Comment by jakethekoenig** [comment]
Oh and one more thing: it was possible to use the new model before this commit. It's just the cost calculations and max context size were wrong.

**Pull request merged**